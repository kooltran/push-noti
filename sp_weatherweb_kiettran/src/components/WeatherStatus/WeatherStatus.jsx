import React from "react";

const WeatherStatus = () => {
  return (
    <div className="weatherStatus-wrapper">
      <div className="weatherStatus-container">
        <div className="weatherStatus-icon"></div>
        <div className="weatherStatus-info">
          <div className="status-text">Cloudy</div>
          <div className="status-info">
            <span className="status-temperature">29.9 C</span>
            <span className="status-humidity">73%</span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default WeatherStatus;
