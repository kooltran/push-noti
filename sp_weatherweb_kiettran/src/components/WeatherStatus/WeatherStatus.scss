.weatherStatus {
  &-wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 38%;
  }

  &-container {
    display: flex;
    justify-content: center;
  }

  &-icon {
    width: 80px;
    height: 80px;
    background-size: contain;
    background-image: url($ICON_CLOUD);
  }

  &-info {
    margin-left: 30px;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
    height: 80px;
    width: 180px;

    .status-text {
      font-size: 30px;
    }

    .status-info {
      display: flex;
      justify-content: space-between;
      font-size: 20px;
    }

    .status-temperature,
    .status-humidity {
      display: flex;
      align-items: center;

      &:before {
        display: inline-block;
        background-image: url($ICON_TEMPERATURE);
        background-size: contain;
        width: 20px;
        height: 20px;
        content: "";
      }
    }

    .status-humidity {
      &:before {
        background-image: url($ICON_HUMIDITY);
      }
    }
  }
}
