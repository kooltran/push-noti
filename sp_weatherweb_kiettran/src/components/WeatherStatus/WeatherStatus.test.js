import React from "react";
import renderer from "react-test-renderer";
import WeatherStatus from "./WeatherStatus";

it("should render WeatherStatus correctly", () => {
  const tree = renderer.create(<WeatherStatus />).toJSON();
  expect(tree).toMatchSnapshot();
});
