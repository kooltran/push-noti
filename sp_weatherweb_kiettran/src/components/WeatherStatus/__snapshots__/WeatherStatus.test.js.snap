// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WeatherStatus correctly 1`] = `
<div
  className="weatherStatus-wrapper"
>
  <div
    className="weatherStatus-container"
  >
    <div
      className="weatherStatus-icon"
    />
    <div
      className="weatherStatus-info"
    >
      <div
        className="status-text"
      >
        Cloudy
      </div>
      <div
        className="status-info"
      >
        <span
          className="status-temperature"
        >
          29.9 C
        </span>
        <span
          className="status-humidity"
        >
          73%
        </span>
      </div>
    </div>
  </div>
</div>
`;
